File: pset6/sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    # input
    text = get_string("Text: ")

    # function calls
    l = count_letters(text)
    w = count_words(text)
    s = count_sentences(text)

    # grading
    grade = 0.0588 * (l * 100 / w) - 0.296 * (s * 100 / w) - 15.8
    grade = round(grade)

    # printing
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade:.2f}")


# count letters
def count_letters(text):
    counter = 0
    for letter in text:
        if letter.isalpha():
            counter += 1
    return counter


# count words
def count_words(text):
    counter = 0
    for letter in text:
        if letter == " ":
            counter += 1

    # adjusting counter
    if len(text) != 0:
        counter += 1
    return counter


# count sentences
def count_sentences(text):
    counter = 0
    for letter in text:
        if letter == "." or letter == "?" or letter == "!":
            counter += 1
    return counter


main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //Getting input
    string text = get_string("Text: ");

    //function calls
    int l = count_letters(text);
    int w = count_words(text);
    int s = count_sentences(text);

    //grading calculation
    float words = w;
    float grade = 0.0588 * (l * 100 / words) - 0.296 * (s * 100 / words) - 15.8;
    grade = round(grade);

    //printing grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", grade);
    }
}

//counting letters
int count_letters(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //check if letter
        if (islower(text[i]) || isupper(text[i]))
        {
            counter++;
        }
    }
    return counter;
}

//counting words
int count_words(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //check if space
        if (text[i] == 32)
        {
            counter++;
        }
    }

    //adjusting counter
    if (strlen(text) != 0)
    {
        counter++;
    }

    return counter;
}

//counting sentences
int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //check if punctuation
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            counter++;
        }
    }
    return counter;
}